pkg/aks/create.go

```diff
@@ -263,84 +150,67 @@ func CreateManagedCluster(ctx context.Context, cred *Credentials, spec *aksv1.AK
 	}
 
 	// Get addon profile from config spec
-	addonProfiles := make(map[string]*containerservice.ManagedClusterAddonProfile, 0)
+	managedCluster.ManagedClusterProperties.AddonProfiles = map[string]*containerservice.ManagedClusterAddonProfile{}
 
 	if hasHTTPApplicationRoutingSupport(spec) {
-		addonProfiles["httpApplicationRouting"] = &containerservice.ManagedClusterAddonProfile{
+		managedCluster.ManagedClusterProperties.AddonProfiles["httpApplicationRouting"] = &containerservice.ManagedClusterAddonProfile{
 			Enabled: spec.HTTPApplicationRouting,
 		}
 	}
 
 	// Get monitoring from config spec
 	if to.Bool(spec.Monitoring) {
-		addonProfiles["omsAgent"] = &containerservice.ManagedClusterAddonProfile{
+		managedCluster.ManagedClusterProperties.AddonProfiles["omsAgent"] = &containerservice.ManagedClusterAddonProfile{
 			Enabled: spec.Monitoring,
 		}
 
-		operationInsightsWorkspaceClient, err := NewOperationInsightsWorkspaceClient(cred)
-		if err != nil {
-			return managedCluster, err
-		}
-
-		logAnalyticsWorkspaceResourceID, err := CheckLogAnalyticsWorkspaceForMonitoring(ctx, operationInsightsWorkspaceClient,
+		logAnalyticsWorkspaceResourceID, err := checkLogAnalyticsWorkspaceForMonitoring(ctx, workplacesClient,
 			spec.ResourceLocation, spec.ResourceGroup, to.String(spec.LogAnalyticsWorkspaceGroup), to.String(spec.LogAnalyticsWorkspaceName))
 		if err != nil {
-			return managedCluster, err
+			return nil, err
 		}
 
 		if !strings.HasPrefix(logAnalyticsWorkspaceResourceID, "/") {
 			logAnalyticsWorkspaceResourceID = "/" + logAnalyticsWorkspaceResourceID
 		}
 		logAnalyticsWorkspaceResourceID = strings.TrimSuffix(logAnalyticsWorkspaceResourceID, "/")
 
-		addonProfiles["omsAgent"].Config = map[string]*string{
+		managedCluster.ManagedClusterProperties.AddonProfiles["omsAgent"].Config = map[string]*string{
 			"logAnalyticsWorkspaceResourceID": to.StringPtr(logAnalyticsWorkspaceResourceID),
 		}
 	}
 
-	// Create a new managed cluster
-	managedCluster = containerservice.ManagedCluster{
-		Name:     to.StringPtr(spec.ClusterName),
-		Location: to.StringPtr(spec.ResourceLocation),
-		Tags:     tags,
-		ManagedClusterProperties: &containerservice.ManagedClusterProperties{
-			KubernetesVersion: spec.KubernetesVersion,
-			AgentPoolProfiles: &agentPoolProfiles,
-			LinuxProfile:      linuxProfile,
-			NetworkProfile:    networkProfile,
-			AddonProfiles:     addonProfiles,
-		},
-	}
-
 	if phase != "updating" && phase != "active" {
 		managedCluster.ServicePrincipalProfile = &containerservice.ManagedClusterServicePrincipalProfile{
 			ClientID: to.StringPtr(cred.ClientID),
 			Secret:   to.StringPtr(cred.ClientSecret),
 		}
 	}
 
-	if spec.DNSPrefix != nil {
+	if to.String(spec.DNSPrefix) != "" {
```

Comment:
Check first `if spec.DNSPrefix != nil {`

Reply:
`to.String()` will always return empty string if the value is nil https://github.com/Azure/go-autorest/blob/main/autorest/to/convert.go#L26